cluster-autoscaler/core/scale_down_test.go

```diff
@@ -315,12 +315,22 @@ func TestScaleDown(t *testing.T) {
 	scaleDown.UpdateUnneededNodes([]*apiv1.Node{n1, n2},
 		[]*apiv1.Node{n1, n2}, []*apiv1.Pod{p1, p2}, time.Now().Add(-5*time.Minute), nil)
 	result, err := scaleDown.TryToScaleDown([]*apiv1.Node{n1, n2}, []*apiv1.Pod{p1, p2}, nil)
+	waitForDeleteToFinish(t, scaleDown)
 	assert.NoError(t, err)
-	assert.Equal(t, ScaleDownNodeDeleted, result)
+	assert.Equal(t, ScaleDownNodeDeleteStarted, result)
 	assert.Equal(t, n1.Name, getStringFromChan(deletedNodes))
 	assert.Equal(t, n1.Name, getStringFromChan(updatedNodes))
 }
 
+func waitForDeleteToFinish(t *testing.T, sd *ScaleDown) {
+	for start := time.Now(); time.Now().Sub(start) < 20*time.Second; time.Sleep(time.Second) {
```

Comment:
I really don't like 20s timeouts in unittests (and 1s polling period too). There is nothing there that should take anywhere near that and having unittests that take more than few seconds will make me stop running them as much as I should while developing stuff.